### PR TITLE
DPL-1000 - Support pacbio request aliquots

### LINKED ISSUE #1159: DPL-1000 - As developers we need to modify requests to use aliquots to support a more flexible way of doing multiplexing
**User story**
As developers we need to modify requests to use aliquots to support a more flexible way of doing multiplexing

**Who are the primary contacts for this story**
James / Steve

**Who is the nominated tester for UAT**
PSD

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] A request can have many aliquots
- [ ] Reception creates primary aliquots for all PacBio requests imported.
- [ ] all existing requests should have a primary aliquot.

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

References
This story has a non-blocking relationship with:
- #1137
- sanger/<repo>#<issue_no.>

**Additional context**

### PR DESCRIPTION
**Pull request by BenTopping**
Closes #1159

#### Changes proposed in this pull request

- Adds rake task to add primary aliquots to requests
- Adds aliquotable concern to requests
- Adds conditional to volume, concentration, template_prep_kit_box_barcode and insert_size presence validation on aliquot to not run for Pacbio::Request aliquots.

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: spec/models/aliquot_spec.rb [hunk 1]
```diff
@@ -3,21 +3,21 @@
 require 'rails_helper'
 
 RSpec.describe Aliquot do
-  it 'is invalid without volume' do
-    expect(build(:aliquot, volume: nil)).not_to be_valid
-  end
+  # it 'is invalid without volume' do
```
Comment (stevieing) [inline spec/models/aliquot_spec.rb:6]: I think it would be better to set these tests to pending with a reason.

File: app/models/aliquot.rb [hunk 2]
```diff
@@ -18,7 +18,7 @@ class Aliquot < ApplicationRecord
   # maybe we need to do this when the state is set to used?
   # we also need to consider that aliquot can be attached to a sample
   # which will not have a barcode or insert size
-  validates :volume, :concentration, :template_prep_kit_box_barcode, :insert_size, presence: true
+  # validates :volume, :concentration, :template_prep_kit_box_barcode, :insert_size, presence: true
```
Comment (stevieing) [inline app/models/aliquot.rb:21]: Is this the best way to do this?

**Review by stevieing — CHANGES_REQUESTED** [review]

File: app/models/aliquot.rb (hunk 2 above)
Comment (BenTopping) [reply app/models/aliquot.rb:21]: Yeah I did wonder if we want to only validate them if source is not on a request until (if) we add request volume tracking. Would that be better?

**Review by stevieing — APPROVED** [review]
Just one question. Will we create a rake task to run all of the migrations together?

**Comment by BenTopping** [comment]
> Just one question. Will we create a rake task to run all of the migrations together?
A single rake task probably isn't needed.
We will do this one on its own because the other migrations rely on requests having a primary aliquot so this one can be run first and doesn't change anything user facing.

**Pull request merged**